internal/ingress/controller/nginx.go

```diff
@@ -991,30 +991,47 @@ func configureBackends(rawBackends []*ingress.Backend) error {
 	return nil
 }
 
+type server struct {
+	Hostname    string `json:"hostname"`
+	Certificate string `json:"certificate"`
+}
+
+type sslConfiguration struct {
+	Certificates map[string]string `json:"certificates"`
+	Servers      map[string]string `json:"servers"`
+}
+
 // configureCertificates JSON encodes certificates and POSTs it to an internal HTTP endpoint
 // that is handled by Lua
 func configureCertificates(rawServers []*ingress.Server) error {
-	servers := make([]*ingress.Server, 0)
+	configuration := &sslConfiguration{
+		Certificates: map[string]string{},
+		Servers:      map[string]string{},
+	}
 
-	for _, server := range rawServers {
-		if server.SSLCert == nil {
+	for _, rawServer := range rawServers {
+		if rawServer.SSLCert == nil {
 			continue
 		}
 
-		servers = append(servers, &ingress.Server{
-			Hostname: server.Hostname,
-			SSLCert: &ingress.SSLCert{
-				PemCertKey: server.SSLCert.PemCertKey,
-			},
-		})
+		uid := rawServer.SSLCert.UID
+		if uid == "" && rawServer.Hostname == defServerName {
+			// empty keys are not valid in lua.
+			uid = "00000000-0000-0000-0000-000000000000"
```

Comment:
can we not make sure default server cert has UID set to this?

Reply:
This only applies to the fakte SSL certificate, i.e, a cert where the source is not a secret. Keep in mind the default certificate could also be a wildcard certificate used in other places.

Reply:
Right, but then we can set this UID when we create the fakesslcert, no?

Reply:
Done